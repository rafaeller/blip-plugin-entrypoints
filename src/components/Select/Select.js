import React, { useRef, useEffect } from 'react';
import PropTypes from 'prop-types';

const Select = ({
    placeholder,
    icon,
    label,
    value = '',
    options = [],
    debounceTime = 500,
    disabled = false,
    onChange = () => {}
}) => {
    const blipSelectRef = useRef(null);
    let time;

    useEffect(() => {
        const { current } = blipSelectRef;
        current.addEventListener('bdsBlur', handleChange);

        return () => {
            current.removeEventListener('bdsBlur', handleChange);
        };
        // eslint-disable-next-line
    }, []);

    const handleChange = (e) => {
        clearTimeout(time);
        time = setTimeout(() => onChange(e), debounceTime);
    };

    return (
        <bds-select
            ref={blipSelectRef}
            icon={icon}
            label={label}
            placeholder={placeholder}
            value={value}
            disabled={disabled}
        >
            {!!options?.length &&
                options.map((option, index) => (
                    <bds-select-option
                        key={`${index}_${option.value}`}
                        value={option.value}
                    >
                        {option.label}
                    </bds-select-option>
                ))}
        </bds-select>
    );
};

Select.propTypes = {
    placeholder: PropTypes.string,
    icon: PropTypes.string,
    label: PropTypes.string,
    value: PropTypes.any,
    options: PropTypes.array,
    debounceTime: PropTypes.number,
    disabled: PropTypes.bool,
    onChange: PropTypes.func
};

export default Select;
