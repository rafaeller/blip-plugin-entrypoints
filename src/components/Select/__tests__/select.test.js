import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import Select from '../Select';

describe('Select component', () => {
    const handleEvent = jest.fn();
    const mockOptions = [
        { value: 'Cat', label: 'Meow' },
        { value: 'Dog', label: 'Woof' }
    ];

    it('should render correctly', () => {
        const { container } = render(
            <Select
                placeholder="Select.."
                value=""
                options={mockOptions}
                onChange={handleEvent}
            />
        );

        expect(container.firstChild).toMatchSnapshot();
    });
});
