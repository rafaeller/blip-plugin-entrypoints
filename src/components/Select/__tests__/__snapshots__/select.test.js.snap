// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render correctly 1`] = `
<bds-select
  disabled="false"
  placeholder="Select.."
  value=""
>
  <bds-select-option
    value="Cat"
  >
    Meow
  </bds-select-option>
  <bds-select-option
    value="Dog"
  >
    Woof
  </bds-select-option>
</bds-select>
`;
